server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -486,6 +490,48 @@ public ActionRequestValidationException validate() {
         return validationException;
     }
 
+    public ActionRequestValidationException validateRescores(ActionRequestValidationException validationException) {
+        RescorerBuilder<?> nonCombinableRescorer = null;
+
+        if (source.rescores() == null) {
+            return validationException;
+        }
+
+        int paginationWindowSize = source.from() + source.size();
+
+        for (RescorerBuilder<?> currentRescorer : source.rescores()) {
+            if (nonCombinableRescorer != null && nonCombinableRescorer.windowSize() < currentRescorer.windowSize()) {
```

Comment:
I am against this API. 

Instead, if we think rescorers should have particular validations, they should do the validations themselves via a `validate` interface that accepts the source. The default implementation could be just return `null`

Reply:
Hey @benwtrent Validation has been moved into the rescorers.